registration/registration_test.go

```diff
@@ -137,6 +140,32 @@ func TestOpeningRounds(t *testing.T) {
 	})
 }
 
+func TestWorkingWithoutWorkerService(t *testing.T) {
+	t.Parallel()
+
+	reg, err := registration.New(
+		context.Background(),
+		time.Now(),
+		t.TempDir(),
+		transport.NewInMemory(),
+		&server.RoundConfig{EpochDuration: time.Millisecond * 10},
+	)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, reg.Close()) })
+
+	var eg errgroup.Group
+	ctx, cancel := context.WithCancel(logging.NewContext(context.Background(), zaptest.NewLogger(t)))
+	defer cancel()
+	eg.Go(func() error { return reg.Run(ctx) })
+
+	for i := 0; i < 3; i++ {
+		round := reg.OpenRound()
+		require.Eventually(t, func() bool { return reg.OpenRound() > round }, time.Second, time.Millisecond*10)
```

Comment:
That's exactly why I didn't want to check if open round increments one by one - to avoid flaky tests. To test the functionality, it just needs to make sure that registration is not stuck (for example on writing to a full channel).

Reply:
OK, but this won't catch other issues like skipped rounds or if rounds have different durations. Maybe a comment clarifying why it is checked like this would help?